### ISSUE TITLE
setHasDeclareKeyword does not work on all ambientable nodes

### ISSUE THREAD

**Issue by kitsonk** [dsherret/ts-simple-ast#460]
**Describe the bug**
Version: 16.0.2

The `.setHasDeclareKeyword()` does not work on all "decarable"/ambientable nodes.

**To Reproduce**

A full reproduction repo is available here: https://github.com/kitsonk/ts-simple-ast-bug

Given the following input in **example.d.ts**:

```ts
export declare const foo: string;
export declare type Foo = string;
export declare interface Bar {
  foo: string;
}
export declare function baz(...args: any[]): void;
```

And operating on it with the following code:

```ts
import { Project, TypeGuards } from "ts-simple-ast";

const project = new Project({
  tsConfigFilePath: "./example/tsconfig.json"
});

const exampleDts = project.getSourceFileOrThrow("example.d.ts");

exampleDts.forEachChild(node => {
  if (TypeGuards.isAmbientableNode(node)) {
    node.setHasDeclareKeyword(false);
  }
});

console.log(exampleDts.print());
```

Will output the following:

```ts
export const foo: string;
export declare type Foo = string;
export declare interface Bar {
    foo: string;
}
export function baz(...args: any[]): void;
```

Where both the type aliases and interfaces still have their `declare` keyword.

**Expected behavior**

Whenever `.setHasDeclareKeyworld(false)` is executed on an ambientable node, it turns off the `declare` keyword on that node.

Refs: denoland/deno#729

**Comment by kitsonk** [comment]
Awesome! 👍 I should have looked at the code... 😁

**Comment by dsherret** [comment]
@kitsonk usually I would take the safe side and consider this kind of bug fix a breaking change, but this was a decision I made probably in the first few days when starting the library that makes no sense. I'm surprised it's been around this long. This is fixed and released in 16.0.3. Thanks for reporting!

**Comment by dsherret** [comment]
Yeah me too haha
